src/firefly/js/Firefly.js

```diff
@@ -278,15 +280,37 @@ function fireflyInit(props, appSpecificOptions={}, webApiCommands) {
         set(appSpecificOptions, 'table.showPropertySheetButton', appSpecificOptions?.table?.showPropertySheetButton ?? true);
     }
 
-    installOptions(appSpecificOptions);
+    const finalOptions= installOptions(appSpecificOptions);
+    const finalAppProps= setupGatorProtocolPanel(finalOptions,appProps);
```

Comment:
I think this approach needs refinement, if not now, then definitely when the next use case arises.  we should avoid adding this type of logic directly in this file.

Reply:
I am not sure where else to add it. I am not sure I want to but in a whole new system for just this.

Reply:
Sure, that's fine for now. However, in the future, I'd like to see an app-options reduction step that processes an ordered list of pure reducing functions (without side effects), where inputs and outputs are clearly defined.

For example, if the menu is affected, it should be passed in as an input and return an updated menu as the output. Any side effects (such as dispatching events) should be handled separately by Firefly.js.

Reply:
That fine... just so you know I did not write `installOptions(appSpecificOptions)` for this PR it has been around for a long time. I just change it to return a value. `setupGatorProtocolPanel` does not have side effects.

But I agree it would be better to modify `installOptions` do it like you said.